adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -864,10 +864,10 @@ private Span newSpan(final String operationName, final Device authenticatedDevic
         if (authenticatedDevice != null && !authenticatedDevice.getDeviceId().equals(sourceAddress.getResourceId())) {
             // gateway scenario
             return getCommandConsumerFactory().createCommandConsumer(sourceAddress.getTenantId(),
-                    sourceAddress.getResourceId(), authenticatedDevice.getDeviceId(), commandHandler, span.context());
+                    sourceAddress.getResourceId(), authenticatedDevice.getDeviceId(), commandHandler, -1, span.context());
```

Comment:
If the parameter is optional, then maybe consider using `Optional`. Or maybe `null` when using `Duration` (see other comments on "int").

Reply:
See the other comment.